### PR TITLE
Refactor: update registrar to use hooks

### PR DESCRIPTION
**Pull request by pauloiankoski**
## Description

This pull request simplifies the process of block registration on the PHP end by replacing the previous registrar process with WP Actions that run at the correct moment for each need. This improves the efficiency and effectiveness of the block registration process.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by kjohnson** [comment]
image

My favorite kind of pull request 😎

**Comment by kjohnson** [comment]
It looks like the failing test is due to a race condition - re-running now.

**Review by kjohnson — COMMENTED** [review]
Looks good to me.

@jonwaldstein can you confirm the recent commit which changes when the scripts are enqueued? Otherwise I would approve this, but want to run it by you.

File: src/NextGen/DonationForm/ViewModels/DonationFormViewModel.php [hunk 1]
```diff
@@ -252,6 +251,8 @@ private function enqueueFormScripts(int $formId, string $formDesignId)
             'give'
         ))->loadInFooter()->enqueue();
 
+        Hooks::doAction('givewp_form_view_enqueue_scripts');
```
Comment (jonwaldstein) [inline src/NextGen/DonationForm/ViewModels/DonationFormViewModel.php:254]: @pauloiankoski this location looks good, but let's change this to:

```suggestion
        Hooks::doAction('givewp_donation_form_enqueue_scripts');
```

**Pull request merged**